tests/test_cli/test_explain_internals.py

```diff
@@ -0,0 +1,64 @@
+"""Unit testing of wps explain command."""
+
+import pytest
+
+from wemake_python_styleguide.cli.commands.explain import (
+    message_formatter,
+    violation_loader,
+)
+from wemake_python_styleguide.violations.best_practices import (
+    InitModuleHasLogicViolation,
+)
+from wemake_python_styleguide.violations.naming import (
+    UpperCaseAttributeViolation,
+)
+from wemake_python_styleguide.violations.oop import BuiltinSubclassViolation
+
+
+@pytest.mark.parametrize(
+    'violation_params',
+    [
+        (115, UpperCaseAttributeViolation),
+        (412, InitModuleHasLogicViolation),
+        (600, BuiltinSubclassViolation),
+    ],
+)
+def test_violation_getter(violation_params):
+    """Test that violation loader can get violation by their codes."""
+    violation_code, expected_class = violation_params
+    violation = violation_loader.get_violation(violation_code)
+    assert violation.code is not None
+    assert violation.docstring == expected_class.__doc__
+
+
+@pytest.mark.parametrize(
+    'test_params',
+    [
+        ('  text\n  text\n  text', 'text\ntext\ntext'),
+        ('  text\n\ttext\r\n  text', 'text\n  text\ntext'),
+        (' text\n    \n\n text', 'text\n\n\ntext'),
+        ('\n\n', '\n\n'),
+        ('text', 'text'),
+        ('text\ntext', 'text\ntext'),
+        ('', ''),
+        ('    ', ''),
+    ],
+)
+def test_indentation_removal(test_params):
```

Comment:
We now don't need to test `textwarp.dedent`, because it is already tested in CPython. And we have integration test that shows that everything is printed fine.